Bug fix for removing stabs info in Symtab::demangleSymbol(), and use
more C++ style.

This patch is temporary until we sync up with dyninst master.


diff -Naurb DyninstAPI-9.0.3.orig/symtabAPI/src/Symtab.C DyninstAPI-9.0.3/symtabAPI/src/Symtab.C
--- DyninstAPI-9.0.3.orig/symtabAPI/src/Symtab.C	2016-01-21 16:15:09.269122103 -0600
+++ DyninstAPI-9.0.3/symtabAPI/src/Symtab.C	2016-01-21 16:21:06.242074446 -0600
@@ -839,10 +839,9 @@
       
 #if !defined(os_windows)        
       //Remove extra stabs information
-      const char *p = strchr(working_name.c_str(), ':');
-      if( p ) {
-         unsigned nchars = p - mangled_name.c_str();
-         working_name = std::string(mangled_name.c_str(), nchars);
+      size_t colon = working_name.find(":");
+      if (colon != std::string::npos) {
+	 working_name = working_name.substr(0, colon);
       }
 #endif        
       
